packages/@aws-cdk/aws-codepipeline/lib/_enhanced_action.ts

```diff
@@ -0,0 +1,36 @@
+import { Aws, ResourceEnvironment } from '@aws-cdk/core';
+import { IAction } from './action';
+
+/**
+ * Helper routines to work with Actions
+ *
+ * Can't put these on Action themselves since we only have an interface
+ * and every library would need to reimplement everything (there is no
+ * `ActionBase`).
```

Comment:
Can we introduce it now and add `env()` as its first member? 
How many such instances are there?

Reply:
Could have, but since we have a ticket for properly refactoring `aws-codepipeline` anyway to support CDKP and other extensions better in the future, I'd rather tackle it as part of that work.